src/index.js

```diff
@@ -39,16 +20,138 @@ const renderList = (list) => {
     const sortedList = list.sort((a, b) => a.index - b.index);
     sortedList.forEach((task) => {
       innerList += `
-        <li class="to-do-tasks">
-          <div class="check-box ${task.completed ? 'completed' : ''}">${
-  task.completed ? '✓' : ''
-}</div>
-          <p class="task-description ${task.completed ? 'line-through' : ''}">${task.description}</p>
-          <img class="more-logo" src="${moreIcon}"/>
-        </li>
-      `;
+          <li class="to-do-tasks" data-task-id="${task.id}">
+            <div class="check-box ${task.completed ? 'completed' : ''}">${task.completed ? '✓' : ''}</div>
+            <p class="task-description ${task.completed ? 'line-through' : ''}" contenteditable>${task.description}</p>
+            <img class="more-logo" src="${moreIcon}" data-task-id="${task.id}"/>
+            <img class="delete-icon" src="${deleteIcon}" alt="Delete" data-task-id="${task.id}" style="width: 16px; height: 16px; display: none;"/>
+          </li>
+          `;
     });
   }
   ToDoListContainer.innerHTML = innerList;
 };
-window.onload = () => renderList(todoList);
+
+const saveToLocalStorage = (list) => {
+  localStorage.setItem('todoList', JSON.stringify(list));
+};
+
+const addTask = (description) => {
+  const newTask = {
+    id: Date.now(),
+    description,
+    completed: false,
+    index: todoList.length + 1,
+  };
+  todoList.push(newTask);
+  saveToLocalStorage(todoList);
+  renderList(todoList);
+};
+
+const deleteTask = (taskId) => {
+  const taskElement = ToDoListContainer.querySelector(`[data-task-id="${taskId}"]`);
+  if (taskElement) {
+    todoList = todoList.filter((task) => task.id !== taskId);
+    todoList.forEach((task, index) => {
+      task.index = index + 1;
+    });
+    saveToLocalStorage(todoList);
+    renderList(todoList);
+  }
+};
+
+const editTaskDescription = (taskId, newDescription) => {
+  todoList.forEach((task) => {
+    if (task.id === taskId) {
+      task.description = newDescription;
+    }
+  });
+  saveToLocalStorage(todoList);
+};
+
+const clearCompletedTasks = () => {
+  todoList = todoList.filter((task) => !task.completed);
+  todoList.forEach((task, index) => {
+    task.index = index + 1;
+  });
+  saveToLocalStorage(todoList);
+  renderList(todoList);
+};
+
+const clearButton = document.querySelector('.clear-btn');
+clearButton.addEventListener('click', () => {
+  clearCompletedTasks();
+});
```

Comment:
- [ ] Let's try to work on project requirements and build our app feature by feature as we're required to. The Clear All Completed feature is not a requirement for this milestone. Kindly remove this functionality as it's meant for a future milestone you'll be working on in the coming tasks. 👍🏽